Unable to Create New Users or Login After User Creation

**Describe the bug**
Following the Various Bug Fixes (#71)  PR being merged I have been unable to create a functional CTF Pad instance.

**To Reproduce**
Steps to reproduce the behavior:
Clone the repository, copy the .env file, and then docker compose up  `docker compose up -d --build`.

Navigate to http://localhost:8000 and create a team.

Using the API key attempt to create a user. On submitting the form with valid data, the following page is displayed.

![image](https://github.com/hugsy/ctfpad/assets/104475159/cc0187d1-0f85-4996-995a-45197574cf0b)

Here is a copy of the full traceback.
```
Environment:


Request Method: POST
Request URL: http://localhost:8000/users/add/

Django Version: 4.2.2
Python Version: 3.11.4
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'django.contrib.sites',
 'model_utils',
 'ctfpad']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 200, in _new_conn
    sock = connection.create_connection(
           
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection
    raise err
    ^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
    ^^^^^^^^^^^^^^^^

The above exception ([Errno 111] Connection refused) was the direct cause of the following exception:
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 790, in urlopen
    response = self._make_request(
               
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 496, in _make_request
    conn.request(
    ^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 388, in request
    self.endheaders()
    ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/http/client.py", line 1281, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/http/client.py", line 1041, in _send_output
    self.send(msg)
    ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/http/client.py", line 979, in send
    self.connect()
    ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 236, in connect
    self.sock = self._new_conn()
                ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 215, in _new_conn
    raise NewConnectionError(
    ^^^^^^^^

The above exception (<urllib3.connection.HTTPConnection object at 0x7f320a8c1910>: Failed to establish a new connection: [Errno 111] Connection refused) was the direct cause of the following exception:
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
           
  File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 844, in urlopen
    retries = retries.increment(
              
  File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 515, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

During handling of the above exception (HTTPConnectionPool(host='localhost', port=3000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f320a8c1910>: Failed to establish a new connection: [Errno 111] Connection refused'))), another exception occurred:
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 184, in post
    return super().post(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 153, in post
    return self.form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/code/ctfpad/views/users.py", line 130, in form_valid
    return super().form_valid(form)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/contrib/messages/views.py", line 12, in form_valid
    response = super().form_valid(form)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/views/generic/edit.py", line 135, in form_valid
    self.object = form.save()
                  ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/forms/models.py", line 542, in save
    self.instance.save()
    ^^^^^^^^^^^^^^^^^^^^
  File "/code/ctfpad/models.py", line 522, in save
    if not register_new_hedgedoc_user(self.hedgedoc_username, self.hedgedoc_password):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/ctfpad/helpers.py", line 66, in register_new_hedgedoc_user
    which_hedgedoc() + '/register',
    ^^^^^^^^^^^^^^^^
  File "/code/ctfpad/helpers.py", line 47, in which_hedgedoc
    requests.get(HEDGEDOC_URL, timeout=CTFPAD_HTTP_REQUEST_DEFAULT_TIMEOUT)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Exception Type: ConnectionError at /users/add/
Exception Value: HTTPConnectionPool(host='localhost', port=3000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f320a8c1910>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

I have confirmed that hedgedoc is accessible from the host on port 3000.

Returning to the login page and attempting to sign in with valid credentials returns the following.

![image](https://github.com/hugsy/ctfpad/assets/104475159/fed21d6e-d8fb-404b-85d1-f35599c3469d)

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/dashboard/

Django Version: 4.2.2
Python Version: 3.11.4
Installed Applications:
['django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.humanize',
 'django.contrib.sites',
 'model_utils',
 'ctfpad']
Installed Middleware:
['django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/ctfpad/decorators/user.py", line 33, in wrapper_func
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/ctfpad/views/__init__.py", line 55, in dashboard
    member = user.member
             ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/utils/functional.py", line 268, in inner
    return func(_wrapped, *args)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/db/models/fields/related_descriptors.py", line 492, in __get__
    raise self.RelatedObjectDoesNotExist(
    ^

Exception Type: RelatedObjectDoesNotExist at /dashboard/
Exception Value: User has no member.
```

**Additional context**
I have not made any changes or additions to the environment file. CTFPad was functional on my machine prior to the merging of PR #71 so I suspect that change is in there. I did also omit the step to `make build` on the most recent version from the main branch since it appears to no longer be required based on the output, `make: *** No rule to make target 'build'.  Stop.`

After the above errors are triggered further attempts to interact with the web app result in various other errors. I can provide tracebacks for those along with any other additional information as needed.

Thanks!

--- a/ctfpad/helpers.py
+++ b/ctfpad/helpers.py
@@ -1,29 +1,34 @@
-from datetime import datetime
-from time import time
-from django.utils.crypto import get_random_string
-import magic
 import os
 import pathlib
-import requests
 import smtplib
-import exrex
-
+from datetime import datetime
 from functools import lru_cache
+from time import time
 from uuid import uuid4
 
+import exrex
+import magic
+import requests
+from django.utils.crypto import get_random_string
+
 from ctftools.settings import (
-    CTFPAD_DOMAIN, CTFPAD_HTTP_REQUEST_DEFAULT_TIMEOUT, CTFPAD_PORT, CTFPAD_USE_SSL,
     CTFPAD_ACCEPTED_IMAGE_EXTENSIONS,
     CTFPAD_DEFAULT_CTF_LOGO,
-    HEDGEDOC_URL,
-    USE_INTERNAL_HEDGEDOC,
-    STATIC_URL,
+    CTFPAD_DOMAIN,
+    CTFPAD_HTTP_REQUEST_DEFAULT_TIMEOUT,
+    CTFPAD_PORT,
+    CTFPAD_USE_SSL,
     CTFTIME_API_EVENTS_URL,
     CTFTIME_USER_AGENT,
-    EMAIL_HOST, EMAIL_HOST_USER, EMAIL_HOST_PASSWORD,
     DISCORD_WEBHOOK_URL,
+    EMAIL_HOST,
+    EMAIL_HOST_PASSWORD,
+    EMAIL_HOST_USER,
     EXCALIDRAW_ROOM_ID_PATTERN,
     EXCALIDRAW_ROOM_KEY_PATTERN,
+    HEDGEDOC_URL,
+    STATIC_URL,
+    USE_INTERNAL_HEDGEDOC,
 )
 
 
@@ -43,11 +48,10 @@ def which_hedgedoc() -> str:
     Returns:
         str: the base HedgeDoc URL
     """
-    try:
-        requests.get(HEDGEDOC_URL, timeout=CTFPAD_HTTP_REQUEST_DEFAULT_TIMEOUT)
-    except ConnectionError:
-        if USE_INTERNAL_HEDGEDOC or HEDGEDOC_URL == 'http://localhost:3000':
-            return 'http://hedgedoc:3000'
+    if USE_INTERNAL_HEDGEDOC:
+        return "http://hedgedoc:3000"
+
+    requests.get(HEDGEDOC_URL, timeout=CTFPAD_HTTP_REQUEST_DEFAULT_TIMEOUT)
     return HEDGEDOC_URL
 
 
@@ -63,9 +67,9 @@ def register_new_hedgedoc_user(username: str, password: str) -> bool:
         bool: if the register action succeeded, returns True; False in any other cases
     """
     res = requests.post(
-        which_hedgedoc() + '/register',
-        data={'email': username, 'password': password},
-        allow_redirects=False
+        which_hedgedoc() + "/register",
+        data={"email": username, "password": password},
+        allow_redirects=False,
     )
 
     if res.status_code != requests.codes.found:
@@ -75,7 +79,7 @@ def register_new_hedgedoc_user(username: str, password: str) -> bool:
 
 
 def create_new_note() -> str:
-    """"Returns a unique note ID so that the note will be automatically created when accessed for the first time
+    """ "Returns a unique note ID so that the note will be automatically created when accessed for the first time
 
     Returns:
         str: the string ID of the new note
@@ -84,7 +88,7 @@ def create_new_note() -> str:
 
 
 def check_note_id(id: str) -> bool:
-    """"Checks if a specific note exists from its ID.
+    """ "Checks if a specific note exists from its ID.
 
     Args:
         id (str): the identifier to check
@@ -119,7 +123,11 @@ def get_file_mime(fpath: pathlib.Path) -> str:
         str: the file mime type, or "application/octet-stream" if the file doesn't exist on FS
     """
     abspath = str(fpath.absolute())
-    return magic.from_file(abspath, mime=True) if fpath.exists() else "application/octet-stream"
+    return (
+        magic.from_file(abspath, mime=True)
+        if fpath.exists()
+        else "application/octet-stream"
+    )
 
 
 def ctftime_parse_date(date: str) -> datetime:
@@ -144,8 +152,8 @@ def ctftime_ctfs(running=True, future=True) -> list:
 
     result = []
     for ctf in ctfs:
-        start = ctf['start']
-        finish = ctf['finish']
+        start = ctf["start"]
+        finish = ctf["finish"]
 
         if running and start < now < finish:
             result.append(ctf)
@@ -164,10 +172,12 @@ def ctftime_fetch_ctfs(limit=100) -> list:
     """
     res = requests.get(
         f"{CTFTIME_API_EVENTS_URL}?limit={limit}&start={time() - (3600 * 24 * 60):.0f}&finish={time() + (3600 * 24 * 7 * 26):.0f}",
-        headers={"user-agent": CTFTIME_USER_AGENT})
+        headers={"user-agent": CTFTIME_USER_AGENT},
+    )
     if res.status_code != requests.codes.ok:
         raise RuntimeError(
-            f"CTFTime service returned HTTP code {res.status_code} (expected {requests.codes.ok}): {res.reason}")
+            f"CTFTime service returned HTTP code {res.status_code} (expected {requests.codes.ok}): {res.reason}"
+        )
 
     result = []
     for ctf in res.json():
@@ -193,7 +203,8 @@ def ctftime_get_ctf_info(ctftime_id: int) -> dict:
     res = requests.get(url, headers={"user-agent": CTFTIME_USER_AGENT})
     if res.status_code != requests.codes.ok:
         raise RuntimeError(
-            f"CTFTime service returned HTTP code {res.status_code} (expected {requests.codes.ok}): {res.reason}")
+            f"CTFTime service returned HTTP code {res.status_code} (expected {requests.codes.ok}): {res.reason}"
+        )
     result = res.json()
     return result
 
@@ -235,13 +246,7 @@ def send_mail(recipients: list, subject: str, body: str) -> bool:
     """
     if EMAIL_HOST and EMAIL_HOST_USER and EMAIL_HOST_PASSWORD:
         try:
-            send_mail(
-                subject,
-                body,
-                EMAIL_HOST_USER,
-                recipients,
-                fail_silently=False
-            )
+            send_mail(subject, body, EMAIL_HOST_USER, recipients, fail_silently=False)
             return True
         except smtplib.SMTPException:
             pass
@@ -294,8 +299,7 @@ def generate_github_page_header(**kwargs) -> str:
     title = kwargs.setdefault("title", "Exported note")
     author = kwargs.setdefault("author", "Anonymous")
     tags = kwargs.setdefault("tags", "[]")
-    date = kwargs.setdefault("date", datetime.now()
-                             ).strftime("%Y-%m-%d %H:%M %Z")
+    date = kwargs.setdefault("date", datetime.now()).strftime("%Y-%m-%d %H:%M %Z")
     content = f"""---
 layout: post
 title: {title}
@@ -320,8 +324,13 @@ def export_challenge_note(member, note_id: uuid4) -> str:
     """
     result = ""
     with requests.Session() as session:
-        h = session.post(f"{HEDGEDOC_URL}/login",
-                         data={"email": member.hedgedoc_username, "password": member.hedgedoc_password})
+        h = session.post(
+            f"{HEDGEDOC_URL}/login",
+            data={
+                "email": member.hedgedoc_username,
+                "password": member.hedgedoc_password,
+            },
+        )
         if h.status_code == requests.codes.ok:
             h2 = session.get(f"{HEDGEDOC_URL}{note_id}/download")
             if h2.status_code == requests.codes.ok:

